docs/reference/modules/discovery/quorums.asciidoc

```diff
@@ -0,0 +1,181 @@
+perform is to elect its first master node, for which it needs to know the set
+of master-eligible nodes whose votes should count in this first election. This
+initial voting configuration is known as the _bootstrap configuration_ and is
+set in the <<modules-discovery-bootstrap-cluster,cluster bootstrapping
+process>>.
+
+It is important that the bootstrap configuration identifies exactly which nodes
+should vote in the first election, and it is not sufficient to configure each
+node with an expectation of how many nodes there should be in the cluster. It
+is also important to note that the bootstrap configuration must come from
+outside the cluster: there is no safe way for the cluster to determine the
+bootstrap configuration correctly on its own.
+
+If the bootstrap configuration is not set correctly then there is a risk when
+starting up a brand-new cluster is that you accidentally form two separate
+clusters instead of one. This could lead to data loss: you might start using
+both clusters before noticing that anything had gone wrong, and it will then be
+impossible to merge them together later.
+
+NOTE: To illustrate the problem with configuring each node to expect a certain
+cluster size, imagine starting up a three-node cluster in which each node knows
+that it is going to be part of a three-node cluster. A majority of three nodes
+is two, so normally the first two nodes to discover each other will form a
+cluster and the third node will join them a short time later. However, imagine
+that four nodes were erroneously started instead of three: in this case there
+are enough nodes to form two separate clusters. Of course if each node is
+started manually then it's unlikely that too many nodes are started, but it's
+certainly possible to get into this situation if using a more automated
+orchestrator, particularly if the orchestrator is not resilient to failures
+such as network partitions.
+
+The initial quorum is only required the very first time a whole cluster starts
+up: new nodes joining an established cluster can safely obtain all the
+information they need from the elected master, and nodes that have previously
+been part of a cluster will have stored to disk all the information required
+when restarting.
+
+[float]
+==== Cluster maintenance, rolling restarts and migrations
+
+Many cluster maintenance tasks involve temporarily shutting down one or more
+nodes and then starting them back up again. By default Elasticsearch can remain
+available if one of its master-eligible nodes is taken offline, such as during a
+<<rolling-upgrades,rolling restart>>. Furthermore, if multiple nodes are stopped
+and then started again then it will automatically recover, such as during a
+<<restart-upgrade,full cluster restart>>. There is no need to take any further
+action with the APIs described here in these cases, because the set of master
+nodes is not changing permanently.
+
+[float]
+==== Auto-reconfiguration
```

Comment:
Maybe "Maintaining the quorum automatically" or something along those lines?

Reply:
Updated in 6af37215ff9 to "Automatic changes to the voting configuration" - I avoided the word "maintaining" because of the collision with "maintenance" in the previous section.